opencog/query/PatternUtils.cc

```diff
@@ -117,21 +116,18 @@ void get_connected_components(
 			size_t nc = components.size();
 			for (size_t i = 0; i<nc; i++)
 			{
-				std::set<Handle> cur_vars(component_vars[i]);
```

Comment:
Ahah, I found the problem! Handle is missing the constructor Handle(std::shared_ptr<const Atom> atom);

instead it only has
```
    explicit Handle(AtomPtr atom);
```

I don't understand enough the inner workings of the atomspace to know if that could be just added, or if it requires some big changes...